src/javascripts/ng-admin/Crud/field/maReferenceField.js

```diff
@@ -0,0 +1,87 @@
+function maReferenceField($compile, ReadQueries) {
+    return {
+        scope: {
+            'field': '&',
+            'value': '=',
+            'entry':  '=?',
+            'datastore': '&?'
+        },
+        restrict: 'E',
+        compile: function() {
+            return {
+                pre: function(scope, element) {
```

Comment:
This is required, as the `link` function would apply on the compiled `ui-select` directive. Yet, I want the custom attributes to be set on the `ui-select` raw element to be sure it propagates them correctly, depending of its custom behavior. For instance, where should we set the `required` attribute on the final ui-select template?

Using `compile` allows to set attributes on the ui-select.